chromium/background.js

```diff
@@ -282,6 +280,15 @@ function onBeforeRequest(details) {
   }
 
   if (httpNowhereOn) {
+    // If loading a main frame, try the HTTPS version as an alternative to
+    // failing.
+    if (shouldCancel) {
+      if (!newuristr) {
+        return {redirectUrl: canonical_url.replace(/^http:/, "https:")};
+      } else {
+        return {redirectUrl: newuristr.replace(/^http:/, "https:")};
+      }
+    }
     if (newuristr && newuristr.substring(0, 5) === "http:") {
```

Comment:
It's actually not: the previous clause (`if (shouldCancel)`) covers the case where the request started with an HTTP request. This clause (`if (newuristr && newuristr.substring(0, 5) === "http:")`) covers the case where we started on an HTTPS page and got redirected to HTTP (by 301 or by a downgrade rule).